contentcuration/contentcuration/frontend/channelEdit/views/ImportFromChannels/SavedSearchesModal.vue

```diff
@@ -1,57 +1,60 @@
-                @click="dialog = false"
+  <div>
+    <KModal
+      v-if="dialog"
+      :title="$tr('savedSearchesTitle')"
+      :cancelText="$tr('closeButtonLabel')"
+      @cancel="dialog = false"
+    >
+      <KCircularLoader v-if="loading" :size="40" />
+      <p v-else-if="savedSearches.length === 0" class="grey--text pa-2">
+        {{ $tr('noSavedSearches') }}
+      </p>
+      <VList v-else>
+        <template v-for="(search, index) in savedSearches">
+          <VListTile :key="index" class="py-2">
+            <VListTileContent>
+              <VListTileTitle>
+                <ActionLink
+                  class="font-weight-bold"
+                  :to="searchResultsRoute(search)"
+                  :text="search.name"
+                  @click="dialog = false"
+                />
+              </VListTileTitle>
+              <VListTileSubTitle class="metadata">
+                <span>
+                  {{ $formatRelative(search.created, { now: new Date() }) }}
+                </span>
+                <span>
+                  {{ $tr('filterCount', { count: searchFilterCount(search) }) }}
+                </span>
+              </VListTileSubTitle>
+            </VListTileContent>
+
+            <VListTileAction>
+              <IconButton
+                icon="edit"
+                color="grey"
+                :text="$tr('editAction')"
+                @click="handleClickEdit(search.id)"
               />
-            </VListTileTitle>
-            <VListTileSubTitle class="metadata">
-              <span>
-                {{ $formatRelative(search.created, { now: new Date() }) }}
-              </span>
-              <span>
-                {{ $tr('filterCount', { count: searchFilterCount(search) }) }}
-              </span>
-            </VListTileSubTitle>
-          </VListTileContent>
-
-          <VListTileAction>
-            <IconButton
-              icon="edit"
-              color="grey"
-              :text="$tr('editAction')"
-              @click="handleClickEdit(search.id)"
-            />
-          </VListTileAction>
-
-          <VListTileAction>
-            <IconButton
-              icon="clear"
-              color="grey"
-              :text="$tr('deleteAction')"
-              @click="handleClickDelete(search.id)"
-            />
-          </VListTileAction>
-        </VListTile>
-        <VDivider v-if="index < savedSearches.length - 1" :key="index + 'divider'" />
-      </template>
-    </VList>
+            </VListTileAction>
+
+            <VListTileAction>
+              <IconButton
+                icon="clear"
+                color="grey"
+                :text="$tr('deleteAction')"
+                @click="handleClickDelete(search.id)"
+              />
+            </VListTileAction>
+          </VListTile>
+          <VDivider v-if="index < savedSearches.length - 1" :key="index + 'divider'" />
+        </template>
+      </VList>
+    </KModal>
 
     <MessageDialog
```

Comment:
I haven't replaced `MessageDialog` and `EditSearchModal` with `KModal` because @marcellamaki already has some work in progress related to that.